src/background.ts

```diff
@@ -1,6 +1,43 @@
+import type { User } from "./types";
+import { createFetcher } from "./utils";
+
 let ext = chrome || browser
 let dashboardUrl = "https://accounts.atoma.cloud/dashboard"
+let authData: User | null = null;
 
 ext.browserAction.onClicked.addListener(() => {
 	window.open(dashboardUrl, "_blank")
-})
+});
+
+const createTimer = () => ext.alarms.create("user-auth-refresh", { delayInMinutes: 1 })
+let atomaTabs: number[] = [];
+
+ext.runtime.onMessage.addListener((message, sender, sendResponse) => {
+	if (message && message.type == "user-auth") {
+		if (sender.tab && sender.tab.id)
+			atomaTabs.push(sender.tab.id)
+		authData = message.user;
+	}
+	sendResponse();
+});
+
+ext.tabs.onRemoved.addListener((tabId) => {
+	atomaTabs = atomaTabs.filter((tab) => tab != tabId)
+});
+
+ext.alarms.onAlarm.addListener((alarm) => {
+	if(alarm.name == "user-auth-refresh") {
+		createTimer();
+		if (authData) {
```

Comment:
Should we have a check if the auth isn't expiring soon then skip?

Reply:
Done